src/controller.rs

```diff
@@ -333,96 +296,9 @@ impl MetricsController {
-                Ok(ref status)=> {
-                    if *status == StatusCode::Ok {
-                        true
-                    } else {
-                        logger().log(LogLevelFilter::Info, "Server said 'not ok' (retry)");
-                        false
-                    }
-                },
-                Err(ref error) => {
-                    logger().log(LogLevelFilter::Error, format!("Error sending data (retry): {}", error).as_str());
-                    false
-                },
-            }) {
-            // This below is the final disposition of retrying n times.
-            Ok(_) => {
-                logger().log(LogLevelFilter::Debug, "Final disposition of 'send': success");
-                return true;
-            },
-            Err(error) => {
-                logger().log(
-                    LogLevelFilter::Error,
-                    format!("Could not send data to server (final): {}", error).as_str()
-                );
-                return false;
-            }
-        }
-    }
-}
-
-#[cfg(test)]
-enum SendResult {
-    Success,
-    Failure
-}
-
 #[cfg(test)]
-struct MockSendWithRetry {
-    retries: u32,
-    wait_time: u32,
-    attempts: u32,
-    succeed_on_attempt: u32,
-    succeeded_on_attempt: u32,
-    result: SendResult
-}
-
-#[cfg(test)]
-impl CanRetry for MockSendWithRetry {
-    fn get_retries(&self) -> u32 { self.retries }
-    fn get_wait_time(&self) -> u32 { self.wait_time }
-    fn send(&mut self) -> Result<StatusCode, String> {
-        //
-        // Should the 'send' function succeed?
-        //
-        match self.result {
-            SendResult::Success => {
-                //
-                // 'send' function should succeed.
-                //
-                // Determine if it should succeed on the current attempt.
-                self.attempts += 1;
-                logger().log(LogLevelFilter::Info,
-                             format!("In MockSendWithRetry::send, attempts: {}", self.attempts).as_str());
-                if self.succeed_on_attempt == self.attempts {
-                    self.succeeded_on_attempt = self.attempts;
-                    logger().log(LogLevelFilter::Info, "In MockSendWithRetry::send, returning Ok (200)");
-                    return Ok(StatusCode::Ok);
-                } else {
-                    // No success yet, return a failure return code
-                    logger().log(LogLevelFilter::Info,
-                                 "In MockSendWithRetry::send, returning Ok (Unauthorized) -- retry");
-                    return Ok(StatusCode::Unauthorized);
-                }
-            },
-            SendResult::Failure => {
-                //
-                // Mock that the 'send' function failed. Return 'Err' object.
-                //
-                return Err("!!!!!! mock error !!!!!!!".to_string());
-            }
-        }
+    pub fn set_telemetry_server_url(&mut self, url: String) {
```

Comment:
Is this for testing?  Or are we exposing this?  If so, then we can have a conflict with the config file.